aiida/tools/importexport/common/config.py

```diff
@@ -178,7 +178,13 @@ def get_all_fields_info():
             'related_name': 'dbnodes'
         },
         'description': {},
-        'process_type': {}
+        'process_type': {},
+        'extras': {
```

Comment:
This changes the export format, right?

I.e. one can now either have export files where the node extras are in `all_fields_info` or where the extras are in `node_extras` (same for attributes).

If that is what we want, we should at the very least document it (but it might make sense to mint a new version, since otherwise we have to support both options in the code).

Reply:
no the node extras are still stored in `node_extras`, same for attributes.
Then currently they are merged in the reader method before yielding the data, i.e. from the interface it does not matter that they are stored separately.
The only difference to the archive file will be these extra entries in the metadata,
but this is currently mitigated (so that you don't need a migration for older files) by already existing lines 383-384 in `aiida/tools/importexport/dbimport/backends/django.py`:

```python
    if entity_name == NODE_ENTITY_NAME:
        # in the current export process these fields are not present,
        # because they are retrieved by a separate query
        fields_info.setdefault('attributes', {'convert_type': 'jsonb'})
        fields_info.setdefault('extras', {'convert_type': 'jsonb'})
```

these lines can be removed when we add a migration for the next version.

Reply:
Ah, I see. Perhaps add a comment to that effect also here in the schema

Reply:
(meaning please add a comment here that these fields are not actually used)

Reply:
> (meaning please add a comment here that these fields are not actually used)

these fields are used though; the writer pops out `node_extras` and `node_attributes` from the fields, before storing to `data.json` (to keep the existing format), but then the reader re-combines them before returning the fields.
The only reason they have to be added here is that, up until this PR, these fields were not stored in fields_info in the archive `metadata.json`.